hbase-server/src/main/java/org/apache/hadoop/hbase/fs/HFileSystem.java

```diff
@@ -85,8 +85,6 @@ public HFileSystem(Configuration conf, boolean useHBaseChecksum)
     this.fs = FileSystem.get(conf);
     this.useHBaseChecksum = useHBaseChecksum;
 
-    fs.initialize(getDefaultUri(conf), conf);
-
```

Comment:
where else do we do the initialize?

Reply:
I'm presuming we need to do it once. I get that `Filesystem.get` should be caching and giving us back the same FileSystem in multiple HFileSystem constructions.

Side note, why are we making multiple instances of HFileSystem?

Reply:
@busbey 
Thanks for having a look. Yes the `FileSystem.get` is giving us the same instance of the Filesystem and always does an intialize also. .So we doing another initialize() is actually creating all member variables more than once. We have seen in our FS impl we have some thread pools - all those are getting created more than once.

Reply:
`  private static FileSystem createFileSystem(URI uri, Configuration conf
      ) throws IOException {

    Class<?> clazz = getFileSystemClass(uri.getScheme(), conf);

    if (clazz == null) {

      throw new IOException("No FileSystem for scheme: " + uri.getScheme());

    }

    FileSystem fs = (FileSystem)ReflectionUtils.newInstance(clazz, conf);

    fs.initialize(uri, conf);

    return fs;

  }`

This is what happens inside `FileSystem.get()`

Reply:
looking at the source and javadocs for hadoop versions we use, I think we should remove this line as you have done and then also replace line 85 with

```
 this.fs = FileSystem.get(getDefaultUri(conf), conf);
```

because the javadocs for that version of get make clear that `initialize` will have already been called on whatever is returned. the get method we're using now is ambiguous about it.

Reply:
Sure @busbey we can change it. but pls do remember that this get() call - what ever flavor of the FS#get() we do, we have already done a call to the FS#get even before we do the HFileSystem creation.
`public static Path getWALRootDir(final Configuration c) throws IOException {

    Path p = new Path(c.get(HBASE_WAL_DIR, c.get(HConstants.HBASE_DIR)));

    if (!isValidWALRootDir(p, c)) {

      return getRootDir(c);

    }

    FileSystem fs = p.getFileSystem(c);

    return p.makeQualified(fs.getUri(), fs.getWorkingDirectory());

  }
`
when we do p.getFileSystem() we have inited the file system.